pytorch_lightning/callbacks/lr_monitor.py

```diff
@@ -259,6 +276,35 @@ def _find_names(self, lr_schedulers: List, add_lr_sch_names: bool = True) -> Lis
 
         return names
 
+    def _find_names_from_optimizer(self, trainer, add_lr_sch_names: bool = True) -> List[str]:
+        names = []
+        seen_optimizers = []
+        seen_optimizer_types = defaultdict(int)
+        for optimizer in trainer.optimizers:
+            name = "lr-" + optimizer.__class__.__name__
+
+            seen_optimizers.append(optimizer)
+            optimizer_cls = type(optimizer)
+            seen_optimizer_types[optimizer_cls] += 1
+
+            # Multiple param groups for the same scheduler
+            param_groups = optimizer.param_groups
+            duplicates = self._duplicate_param_group_names(param_groups)
+            if duplicates:
+                raise MisconfigurationException(
+                    "A single `Optimizer` cannot have multiple parameter groups with identical "
+                    f"`name` values. {name} has duplicated parameter group names {duplicates}"
+                )
+
+            name = self._add_prefix(name, optimizer_cls, seen_optimizer_types)
+
+            names.extend(self._add_suffix(name, param_groups, i) for i in range(len(param_groups)))
+
+            if add_lr_sch_names:
+                self.lr_sch_names.append(name)
+
```

Comment:
same here. there's a lot of duplicated code above. let's try to abstract these into common helper functions

Reply:
I have added helper methods.